src/lib.rs

```diff
@@ -22,8 +59,50 @@ use std::fs;
 use std::path::{self, PathBuf, Path};
 use rand::{thread_rng, Rng};
 
-/// A wrapper for a path to temporary directory implementing automatic
-/// scope-based deletion.
+/// A directory in the filesystem that is automatically deleted when
+/// it goes out of scope.
+///
+/// The [`TempDir`] type creates a directory on the file system that
+/// is deleted once it goes out of scope. At construction, the
+/// `TempDir` creates a new directory with a randomly generated name,
+/// and with a prefix of your choosing.
+///
+/// The default constructor, [`TempDir::new`], creates directories in
+/// the location returned by [`std::env::temp_dir()`], but they can be
+/// configured to manage the lifetime of a temporary directory in any
+/// location by constructing with [`TempDir::new_in`].
+///
+/// After creating a `TempDir`, work with the file system by doing
+/// standard [`std::fs`] file system operations on it's [`Path`],
+/// which can be retrieved with [`TempDir::path`]. Once the `TempDir`
```

Comment:
Could we maybe say `TempDir.path`? I see `TempDir::path` and instantly think associated functions and not ones that operate on references. I am not sure if this is a standard thing to do or not just a thought.